docs/spec/helper.md

```diff
@@ -0,0 +1,90 @@
+# Implementation Helper
+
+This document supplements the official [spec](spec.md) with useful
+implementation details.
```

Comment:
If this only pertains to the ChannelBased Broker, then I think we should state that up front so it's clear which implementation details we're talking about. Otherwise the conditions, for example L40 should be matched with the spec.